packages/opentelemetry-exporter-collector/src/platform/browser/CollectorExporter.ts

```diff
@@ -120,7 +120,9 @@ export class CollectorExporter extends CollectorExporterBase<
     xhr.setRequestHeader('Accept', 'application/json');
     xhr.setRequestHeader('Content-Type', 'application/json');
     Object.entries(this._headers).forEach(([k, v]) => {
-      xhr.setRequestHeader(k, v);
+      if (typeof v !== 'undefined') {
```

Comment:
Does it make sense to validate the headers during initialization rather than doing this check on each export? If so, logging that a header was invalid and won't be sent could be helpful.

Reply:
hmm good point , everything was fine until I added Partial to definition of headers. and I think I will remove it as it causes ts compiler to complain in this place